pkg/agent/flexvolume/crd/register.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package pool to manage a rook pool.
+package crd
+
+import (
+	"reflect"
+
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"github.com/rook/rook/pkg/operator/kit"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	CustomResourceName       = "volumeattach"
+	CustomResourceNamePlural = "volumeattachs"
```

Comment:
I named it like that to match this arg: https://github.com/rook/rook/blob/master/pkg/operator/kit/resource.go#L44

I also wanted to emphasize that this is the CRD Name during registration. Using a variable `volumeattachment` doesnt say that it is the CRD name.

Reply:
I don't understand your argument.  I do understand that the CRD name needs a singular and plural version, but as it is now the plural doesn't make english sense.

For singular/plural, either use `volumeattach` and `volumeattaches`, or `volumeattachment` and `volumeattachments`.

`volumeattachs` isn't proper english.

Reply:
ahh that is what you meant. You were talking about the values and not the keys. . This was done to make it work with TPR because in TPR, for querying the rest API, it adds an 's' at the end. So GET /api/1/volumeattachs/123 etc.

I can rename it to volumeattachment/volumeattachments